base/vflip/vflip.man
.ds Vn 1.12
.TH vflip 1Vi "24 April 1993" "Vista Version \*(Vn"
.SH NAME
vflip \- flip an image horizontally and/or vertically
.SH SYNOPSIS
\fBvflip\fR [\fB-hor\fR] [\fB-vert\fR] [\fB-\fIoption\fR ...] \
[\fIinfile\fR] [\fIoutfile\fR]
.SH DESCRIPTION
\fBvflip\fP flips images horizontally and/or vertically.
.PP
A horizontal flip exchanges the left and right sides of an image. A 
vertical flip exchanges the top and bottom. 
.SH "COMMAND LINE OPTIONS"
\fBvflip\fP accepts the following options:
.IP \fB-help\fP 15n
Prints a message describing options.
.IP "\fB-in\fP \fIinfile\fP"
Specifies a Vista data file containing the input images.
.IP "\fB-out\fP \fIoutfile\fP"
Specifies where to write the flipped images as a Vista data file.
.IP "\fB-horizontal\fP"
Specifies that a horizontal flip should be performed.
.IP "\fB-vertical\fP"
Specifies that a vertical flip should be performed.
.PP
If both \fB-horizontal\fP and \fB-vertical\fP are specified, the horizontal 
flip is performed first. If neither is specified, the images are 
nevertheless flipped horizontally. 
.SH "SEE ALSO"
.na
.nh
.BR vrotate (1Vi),
.BR vtranspose (1Vi),
.BR VImage (3Vi),
.BR Vista (7Vi)
.ad
.hy
.SH AUTHOR
Daniel Ko <ko@cs.ubc.ca>


debian/changelog
via (2.0.1svn191-0) stable; urgency=low

  *renamed via-bin to via-utils

 -- Lipsia Group (The official Lipsia developer group) <lipsia@cbs.mpg.de>  Tue, 09 Nov 2010 12:17:00 +0400 

via (2.0.0-0) stable; urgency=low

  *This version comes with the new lipsia-2.0 package
   and comprises some minor bug fixes
  *Fixed type error of VLong (is now long)

 -- Lipsia Group (The official Lipsia developer group) <lipsia@cbs.mpg.de>  Tue, 09 Nov 2010 12:17:00 +0400 

via (1.6.0-2) unstable; urgency=low

  * New internal release

 -- Frank Burkhardt <burk@cbs.mpg.de>  Mon, 20 Jul 2009 11:45:41 +0200

via (1.6.0-1) unstable; urgency=low

  * Initial release (Closes: #521328).
  * New Upstream Version. Renamed 'vcat' to 'vcatobj' to not conflict with
    atfs package in Debian.
  * Update Standards-version to 3.8.2.0 -- no changes necessary.
  * Update watchfile for recent change in upstream naming scheme.
  * Add versioned dependency on debhelper (>= 5) upon recommendation by
    lintian.

 -- Michael Hanke <michael.hanke@gmail.com>  Fri, 19 Jun 2009 13:50:52 +0200

via (1.5.2+svn3529-2) UNRELEASED; urgency=low

  * Add alternative build-dependency on atlas3-base-dev to be able to build
    the package on Debian etch.
  * FTP access to the upstream sources needs passive mode. Fixed debian/watch.

 -- Michael Hanke <michael.hanke@gmail.com>  Tue, 14 Apr 2009 02:46:10 +0200

via (1.5.2+svn3529-1) UNRELEASED; urgency=low

  * New Upstream SVN snapshot that relocates the VIA lib headers to their new
    cannonical location.

 -- Michael Hanke <michael.hanke@gmail.com>  Tue, 31 Mar 2009 22:08:01 +0200

via (1.5.2-1) UNRELEASED; urgency=low

  * Initial Debian packaging. This is a complete rewrite of the original
    Debian packaging effort done by the upstream developers.

 -- Michael Hanke <michael.hanke@gmail.com>  Thu, 26 Mar 2009 19:46:09 +0100


base/vflip/vflip.c
/****************************************************************
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 *
 * $Id: vflip.c 3629 2009-08-20 17:04:30Z proeger $
 *
 *****************************************************************/

/*! \brief vflip - performs one or more flip operations

\par Description
vflip flips images horizontally and/or vertically.
<p>
A  horizontal  flip  exchanges the left and right sides of an image. A vertical flip
exchanges the top and bottom.

\par Usage

        <code>vflip</code>

        \param -in          input image
        \param -out         output image
        \param -horizontal  Flip horizontally. Default: false
        \param -vertical    Flip vertically. Default: false

\par Examples
<br>
\par Known bugs
none.

\file vflip.c
\author Daniel Ko, UBC Laboratory for Computational Intelligence
*/


/*
 *  The vflip program, which performs one or more flip operations
 *  on Vista images.
 */

/*
 *  Copyright 1993, 1994 University of British Columbia
 *
 *  Permission to use, copy, modify, distribute, and sell this software and its
 *  documentation for any purpose is hereby granted without fee, provided that
 *  the above copyright notice appears in all copies and that both that
 *  copyright notice and this permission notice appear in supporting
 *  documentation. UBC makes no representations about the suitability of this
 *  software for any purpose. It is provided "as is" without express or
 *  implied warranty.
 *
 *  Author: Daniel Ko, UBC Laboratory for Computational Intelligence
 */

/* From the Vista library: */
#include "viaio/Vlib.h"
#include "viaio/mu.h"
#include "viaio/option.h"
#include "viaio/os.h"
#include "viaio/VImage.h"



int main (int argc, char **argv)
{
    static VBoolean horizontal_flip = FALSE, vertical_flip = FALSE;
    static VOptionDescRec options[] = {
	{ "horizontal", VBooleanRepn, 1, & horizontal_flip, VOptionalOpt, NULL,
	      "Flip horizontally" },
	{ "vertical", VBooleanRepn, 1, & vertical_flip, VOptionalOpt, NULL,
	      "Flip vertically" }
    };
    FILE *in_file, *out_file;
    VAttrList list;
    VAttrListPosn posn;
    VImage image;
    int nimages = 0;
    char prg[50];	
    sprintf(prg,"vflip V%s", getVersion());
    fprintf (stderr, "%s\n", prg);


    /* Parse command line arguments: */
    VParseFilterCmd (VNumber (options), options, argc, argv,
		     & in_file, & out_file);

    /* Default to horizontal flip if no flipping is specified: */
    if (! horizontal_flip && ! vertical_flip)
	horizontal_flip = TRUE;

    /* Read source image(s): */
    if (! (list = VReadFile (in_file, NULL)))
	exit (EXIT_FAILURE);
    
    /* Perform flip operations on each image: */
    for (VFirstAttr (list, & posn); VAttrExists (& posn); VNextAttr (& posn)) {
	
	if (VGetAttrRepn (& posn) != VImageRepn)
	    continue;
	VGetAttrValue (& posn, NULL, VImageRepn, & image);
	
	/* First flip horizontal: */
	if (horizontal_flip && ! VFlipImage (image, image, VAllBands, FALSE))
	    exit (EXIT_FAILURE);
	
	/* Then flip vertical: */
	if (vertical_flip && ! VFlipImage (image, image, VAllBands, TRUE))
	    exit (EXIT_FAILURE);

	nimages++;
    }

    /* Write the results to the output file: */
    if (! VWriteFile (out_file, list))
	exit (EXIT_FAILURE);

    fprintf (stderr, "%s: Flipped %d image(s) %s%s%s.\n",
	     argv[0], nimages,
	     horizontal_flip ? "horizontally" : "",
	     (horizontal_flip && vertical_flip) ? " and " : "",
	     vertical_flip ? "vertically" : "");

    return EXIT_SUCCESS;
}
